src/Common/src/System/Net/SafeCloseSocket.Unix.cs

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using Microsoft.Win32.SafeHandles;
+using System.Diagnostics;
+
+namespace System.Net.Sockets
+{
+#if DEBUG
+    internal sealed partial class SafeCloseSocket : DebugSafeHandleMinusOneIsInvalid
+#else
+    internal sealed partial class SafeCloseSocket : SafeHandleMinusOneIsInvalid
+#endif
+    {
+        public SocketAsyncContext AsyncContext
+        {
+            get
+            {
+                return _innerSocket.AsyncContext;
+            }
+        }
+
+		public int FileDescriptor
+		{
+			get
+			{
+				return (int)handle;
+			}
+		}
+
+        public ThreadPoolBoundHandle IOCPBoundHandle
+        {
+            get
+            {
+                // TODO: remove this once async sockets are PAL'd out
+                throw new PlatformNotSupportedException();
+            }
+        }
+
+        public unsafe static SafeCloseSocket CreateSocket(int fileDescriptor)
+        {
+            return CreateSocket(InnerSafeCloseSocket.CreateSocket(fileDescriptor));
+        }
+
+		public unsafe static SafeCloseSocket CreateSocket(AddressFamily addressFamily, SocketType socketType, ProtocolType protocolType)
+		{
+			return CreateSocket(InnerSafeCloseSocket.CreateSocket(addressFamily, socketType, protocolType));
+		}
+
+		public unsafe static SafeCloseSocket Accept(SafeCloseSocket socketHandle, byte[] socketAddress, ref int socketAddressSize)
+		{
+			return CreateSocket(InnerSafeCloseSocket.Accept(socketHandle, socketAddress, ref socketAddressSize));
+		}
+
+        public ThreadPoolBoundHandle GetOrAllocateThreadPoolBoundHandle()
+        {
+            // TODO: remove this once async sockets are PAL'd out
+            throw new PlatformNotSupportedException();
+        }
+
+        private void InnerReleaseHandle()
+        {
+			// No-op for Unix.
+        }
+
+        internal sealed partial class InnerSafeCloseSocket : SafeHandleMinusOneIsInvalid
+        {
+            private SocketAsyncContext _asyncContext;
+
+            public SocketAsyncContext AsyncContext
+            {
+                get
+                {
+                    if (Volatile.Read(ref _asyncContext) == null)
+                    {
+                        Interlocked.CompareExchange(ref _asyncContext, new SocketAsyncContext((int)handle, SocketAsyncEngine.Instance), null);
```

Comment:
There is not, no (just the opposite, in fact: the losers must not remove the socket fd from the epoll instance).